distributed/protocol/utils.py

```diff
@@ -89,10 +89,7 @@ def merge_frames(header, frames):
 
 
 def pack_frames_prelude(frames):
-    lengths = [struct.pack("Q", len(frames))] + [
-        struct.pack("Q", nbytes(frame)) for frame in frames
-    ]
-    return b"".join(lengths)
+    return struct.pack(f"Q{len(frames)}Q", len(frames), *map(nbytes, frames))
```

Comment:
With more complicated comprehensions I sometimes have trouble reading what it does .  Maybe include a docstring like:
> Pack frames into bytestring of length of frames and size of each frame

Maybe it's fine...if you thinks it's clear then merge away

Reply:
Assigned a couple variables beforehand. Please let me know if that is clearer.

Reply:
Thanks @jakirkham